cabal-install/Distribution/Client/CmdUpdate.hs

```diff
@@ -0,0 +1,179 @@
+{-# LANGUAGE CPP, NamedFieldPuns, RecordWildCards, ViewPatterns #-}
+
+-- | cabal-install CLI command: update
+--
+module Distribution.Client.CmdUpdate (
+    updateCommand,
+    updateAction,
+  ) where
+
+import Distribution.Client.ProjectOrchestration
+import Distribution.Client.ProjectConfig
+         ( ProjectConfig(..)
+         , projectConfigWithSolverRepoContext )
+import Distribution.Client.Types
+         ( Repo(..), RemoteRepo(..), isRepoRemote )
+import Distribution.Client.HttpUtils
+         ( DownloadResult(..) )
+import Distribution.Client.FetchUtils
+         ( downloadIndex )
+import Distribution.Client.JobControl
+         ( newParallelJobControl, spawnJob, collectJob )
+import Distribution.Client.Setup
+         ( GlobalFlags, ConfigFlags(..), ConfigExFlags, InstallFlags, UpdateFlags (updateIndexState)
+         , applyFlagDefaults, defaultUpdateFlags, RepoContext(..) )
+import Distribution.Simple.Setup
+         ( HaddockFlags, fromFlagOrDefault, fromFlag )
+import Distribution.Simple.Utils
+         ( die', notice, wrapText, writeFileAtomic, noticeNoWrap, intercalate )
+import Distribution.Verbosity
+         ( Verbosity, normal, lessVerbose )
+import Distribution.Client.IndexUtils.Timestamp
+import Distribution.Client.IndexUtils
+         ( updateRepoIndexCache, Index(..), writeIndexTimestamp
+         , currentIndexTimestamp )
+import Distribution.Text
+         ( Text(..), display, simpleParse )
+
+import qualified Distribution.Compat.ReadP  as ReadP
+import qualified Text.PrettyPrint          as Disp
+
+import Control.Monad (unless, when)
+import qualified Data.ByteString.Lazy       as BS
+import Distribution.Client.GZipUtils (maybeDecompress)
+import System.FilePath (dropExtension)
+import Data.Time (getCurrentTime)
+import Distribution.Simple.Command
+         ( CommandUI(..), usageAlternatives )
+import qualified Distribution.Client.Setup as Client
+
+import qualified Hackage.Security.Client as Sec
+
+updateCommand :: CommandUI (ConfigFlags, ConfigExFlags, InstallFlags, HaddockFlags)
+updateCommand = Client.installCommand {
+  commandName         = "new-update",
+  commandSynopsis     = "Updates list of known packages.",
+  commandUsage        = usageAlternatives "new-update" [ "[FLAGS]" ],
+  commandDescription  = Just $ \_ -> wrapText $
+        "For all known remote repositories, download the package list.",
+  commandNotes        = Just $ \pname ->
+        "Examples:\n"
+     ++ "  " ++ pname ++ " new-update\n"
+     ++ "    Download the package list for all known remote repositories.\n\n"
```

Comment:
We may want to add an example here showing how to update a single repo

and we may point out that a `TARGET` has the format `<repo-id>[,<index-state>]` where `<index-state>` is the same syntax as supported by the `--index-state` flag which currently is:

```
    --index-state=STATE              Use source package index state as it
                                     existed at a previous time. Accepts
                                     unix-timestamps (e.g. '@1474732068'),
                                     ISO8601 UTC timestamps (e.g.
                                     '2016-09-24T17:47:48Z'), or 'HEAD'
                                     (default: 'HEAD').
```